### PR TITLE
Add projectProperties for bulk property addition

### PR DESCRIPTION
**Pull request by lukebemish**

### CODE REVIEW

File: gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy [hunk 1]
```diff
@@ -79,6 +79,16 @@ abstract class AbstractGatherPlatformDetailsTask extends DefaultTask {
         )
     }
 
+    void projectProperties(Collection<String> names) {
+        names.each { projectProperty(it) }
```
Comment (PaintNinja) [inline gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy:83]:
```suggestion
        names.forEach(this::projectProperty)
```

File: gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy [hunk 2]
```diff
@@ -79,6 +79,16 @@ abstract class AbstractGatherPlatformDetailsTask extends DefaultTask {
...
+    }
+
+    void projectProperties(Provider<? extends Collection<String>> names) {
+        buildProperties.putAll(names.<Map<String, Object>>map { Collection<String> it ->
+            it.collectEntries { [(it):project.property(it)] }
```
Comment (PaintNinja) [inline gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy]: Minor formatting nitpick
```suggestion
            it.collectEntries { [(it): project.property(it)] }
```

File: gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy [hunk 3]
```diff
@@ -87,6 +97,18 @@ abstract class AbstractGatherPlatformDetailsTask extends DefaultTask {
         buildProperties.putAll(project.providers.gradleProperty(name).<Map<String, Object>>map { it -> [(name): it] })
     }
 
+    void gradleProperties(Collection<String> names) {
+        names.each { gradleProperty(it) }
```
Comment (PaintNinja) [inline gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy:101]:
```suggestion
        names.forEach(this::gradleProperty)
```

**Review by PaintNinja — CHANGES_REQUESTED** [review]

File: gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy (hunk 1 above)
Comment (lukebemish) [reply gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy:83]: Yeah this one I'm not sold on; I feel like we should use the groovy-specific `each` where possible, and the closures really aren't adding _that_ much space to the jar.

Comment (lukebemish) [reply gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/tasks/AbstractGatherPlatformDetailsTask.groovy:83]: Turns out a method pointer (`this.&projectProperty`) reflects the class each invocation in groovy 3 to get the method so... yeah, just leaving this as the closure

**Pull request merged**